editor/scene_tree_dock.cpp

```diff
@@ -1415,6 +1421,82 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 					}
 					undo_redo->add_do_method(node, "set_unique_name_in_owner", false);
 					undo_redo->add_undo_method(node, "set_unique_name_in_owner", true);
+					if (node->is_exposed_in_owner()) {
+						undo_redo->add_undo_method(node, "set_exposed_in_owner", true);
+					}
+				}
+				undo_redo->commit_action();
+			}
+		} break;
+		case TOOL_TOGGLE_SCENE_EXPOSED: {
```

Comment:
There is so much code duplication here that it feels like it should somehow be shared with `TOOL_TOGGLE_SCENE_UNIQUE_NAME`. But that's something that can be improved later.

Reply:
agreed, ill explore this